src/unix/core.c

```diff
@@ -1284,6 +1288,70 @@ int uv_translate_sys_error(int sys_errno) {
 }
 
 
+int uv_os_environ(uv_env_item_t** envitems, int* count) {
+  int i, j, cnt;
+  uv_env_item_t* envitem;
+
+  *envitems = NULL;
+  *count = 0;
+
+  for (i = 0; environ[i]; i++);
+
+  *envitems = uv__calloc(i, sizeof(**envitems));
+
+  if (envitems == NULL)
+    return UV_ENOMEM;
+
+  for (j = 0, cnt = 0; j < i; j++) {
+    char* buf;
+    char* ptr;
+
+    buf = uv__strdup(environ[j]);
+    if (buf == NULL)
+      goto fail;
+
+    ptr = strchr(buf, '=');
+    if (ptr == NULL) {
```

Comment:
Can this even happen?

Reply:
I honestly don't know.

Reply:
@libuv/collaborators any suggestions on what to do here? I'm not sure this is possible, but I chose to ignore those bogus entries, if present.

Reply:
`environ` is freely mutable by the program so yes, you should be prepared for bogus entries. `getenv()` also ignores such entries so I think it's fine to do that in libuv as well.

Reply:
BTW, I don't know either and I'm ok with this code.